server/channels/app/platform/web_conn.go

```diff
@@ -843,7 +869,18 @@ func (wc *WebConn) ShouldSendEvent(msg *model.WebSocketEvent) bool {
 	}
 
 	// Only report events to users who are in the channel for the event
-	if msg.GetBroadcast().ChannelId != "" {
+	if chID := msg.GetBroadcast().ChannelId; chID != "" {
+		// For typing events, we don't send them to users who don't have
+		// that channel or thread opened.
+		if wc.Platform.Config().FeatureFlags.WebSocketEventScope &&
+			utils.Contains([]model.WebsocketEventType{
+				model.WebsocketEventTyping,
+				model.WebsocketEventReactionAdded,
+				model.WebsocketEventReactionRemoved,
+			}, msg.EventType()) && wc.notInChannel(chID) && wc.notInThread(chID) {
```

Comment:
What would happen if a user switches to a channel right in the moment a reaction gets added. Would this make the client "vulnerable" to race conditions that could lead to missing the reaction until refresh?

Reply:
Switching to a channel will always give you the reaction. Irrespective of whether you miss the ws event or not. As long as we handle the case of the event being sent when the user is already on a channel, it should be fine.

Reply:
Right but let's say the reaction is added after the data load operation due to switching channels happened but before client is able to set the active channel. Not sure if these edge cases are completely solvable with a simple fire and forget websocket event.

Reply:
Oh I see. That should be solvable by ensuring the data load happens after the active channel is set. I'll check and confirm that.

Reply:
Yeah, that would fix one side of this potential problem, but still there's technically no guarantee that the server has processed the client message and set the id unless we send a response. I understand it's highly unlikely but a message could be delayed enough for the race to trigger even if the data ordering is proper.

Reply:
Well, we do send a response. But then tracking and retrying that starts to get complicated. I think this is a good callout and will mention this in the doc. :+1:

Reply:
Yep, confirmed that's the case. Will update the design doc to reflect this discussion.